ClimateEmergency_Engine/Compute/LCA/EvaluateEPD.cs

```diff
@@ -36,39 +37,52 @@ public static partial class Compute
+        [Input("epdData", "BHoM Data object containing Environmental Product Declaration metrics.")]
         [Input("epdField", "Text string corresponding to one of the available fields in the epdData set.")]
         [Output("quantity", "The effect of the EPD field specified for this object")]
-        public static double EvaluateEPD(BHoMObject obj, CustomObject epdData, string epdField)
+        public static double EvaluateEPD(BHoMObject obj = null, CustomObject epdData = null, EPDField epdField = EPDField.GlobalWarmingPotential)
         {
-            double mass = IMass(obj);
+            return IMass(obj) * CompileEPD(epdData, epdField);
+        }
 
+        public static double CompileEPD(CustomObject epdData, EPDField epdField)
+        {
             double epdValue = 0;
 
-            if (epdData.CustomData.ContainsKey(epdField))
+            if (epdData.CustomData.ContainsKey(System.Convert.ToString(epdField)))
             {
                 try
                 {
-                    epdValue = System.Convert.ToDouble(epdData.CustomData[epdField]);
+                    epdValue = System.Convert.ToDouble(epdData.CustomData[System.Convert.ToString(epdField)]);
                 }
                 catch
                 {
                     BH.Engine.Reflection.Compute.RecordWarning("The value for {epdField} is not a number, or is not valid.");
                     return 0;
-                }                    
+                }
             }
             else
             {
                 BH.Engine.Reflection.Compute.RecordError($"The EPDDataset must have a value for {epdField}");
                 return 0;
             }
-
-            return mass * epdValue;
+            return epdValue;
         }
 
         /***************************************************/
 
+        [Description("Calculates the acidification potential of a BHoM Object based on explicitly defined volume and Environmental Product Declaration dataset.")]
+        [Input("volume", "Volume as a double. This method does not extract the Volume of an object and should be provided manually. The property may be extracted from a BHoM EPD Dataset.")]
+        [Input("density", "Density as a double. This method does not extract the Density of an object and should be provided manually. The property may be extracted from a BHoM EPD Dataset.")]
+        [Input("epdData", "BHoM Data object containing values for typical metrics within Environmental Product Declarations.")]
+        [Input("epdField", "BHoM EPDField Enum to select Environmental Product Declaration metric for evaluation.")]
+        [Output("quantity", "The effect of the EPD field specified for this object")]
+
+        public static double EvaluateEPD(double volume = 0, double density = 0, CustomObject epdData = null, EPDField epdField = EPDField.GlobalWarmingPotential)
+        {
+            return (volume * density) * CompileEPD(epdData, epdField);
```

Comment:
Could add:
`density = (density == 0) ? CompileEPD(epdData, EPDField.Density : density`
to get the density from the EPD if user doesn't specify.